apps/web/src/stores/datasets.ts

```diff
@@ -137,6 +139,40 @@ export default function useDatasets(
       },
     })
 
+  const {
+    execute: runGenerateAction,
+    isPending: generateIsLoading,
+    error: generateError,
+  } = useLatitudeAction(generateDatasetAction, {
+    onError: (error) => {
+      toast({
+        title: 'Failed to generate dataset',
+        description: error.message,
+        variant: 'destructive',
+      })
+    },
+    onSuccess: ({ data: dataset }) => {
+      mutate([...data, dataset])
+    },
+  })
+
+  const {
```

Comment:
Same. This could be in the onboarding folder. No need to make globally available